api/server/creds.go

```diff
@@ -11,61 +12,155 @@ import (
 func (vd *volAPI) credsEnumerate(w http.ResponseWriter, r *http.Request) {
 	method := "credsEnumerate"
 
-	d, err := vd.getVolDriver(r)
+	ctx, err := vd.annotateContext(r)
+
 	if err != nil {
-		notFound(w, r)
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	creds, err := d.CredsEnumerate()
+	// Get gRPC connection
+	conn, err := vd.getConn()
 	if err != nil {
 		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	json.NewEncoder(w).Encode(creds)
+
+	client := api.NewOpenStorageCredentialsClient(conn)
+	resp, err := client.Enumerate(ctx, &api.SdkCredentialEnumerateRequest{})
+
+	if err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	credentials := make(map[string]interface{})
+
+	for _, credentialId := range resp.CredentialIds {
+		credResp, err := client.Inspect(r.Context(), &api.SdkCredentialInspectRequest{
+			CredentialId: credentialId,
+		})
+
+		if err != nil {
+			vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		credentials[credResp.CredentialId] = credResp
+	}
+
+	if err := json.NewEncoder(w).Encode(credentials); err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
 }
 
 func (vd *volAPI) credsCreate(w http.ResponseWriter, r *http.Request) {
 	method := "credsCreate"
-	var input api.CredCreateRequest
-	response := &api.CredCreateResponse{}
+	var (
+		input api.CredCreateRequest
+		err   error
+	)
 
-	if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
+	if err = json.NewDecoder(r.Body).Decode(&input); err != nil {
+		vd.sendError(vd.name, method, w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	useProxy := false
+
+	if len(input.InputParams[api.OptCredProxy]) != 0 {
+		useProxy, err = strconv.ParseBool(input.InputParams[api.OptCredProxy])
+
+		if err != nil {
+			vd.sendError(vd.name, method, w, err.Error(), http.StatusBadRequest)
+			return
+		}
+	}
+
+	ownerShip, err := getOwnershipFromString(input.InputParams[api.OptCredOwnership])
```

Comment:
nit: ownership is one word

Reply:
You can remove this from the REST call. This was not supported through the old REST server. It is only for the SDK